6-infrastructure-configuration-management/6.1.3-terraform-modules.md

```diff
@@ -4,6 +4,192 @@ The previous sections should have illustrated how flexible and powerful Terrafor
 
 Read Hashicorp's [Terraform module overview](https://developer.hashicorp.com/terraform/tutorials/modules/module) to learn more about Terraform modules features and uses.
 
+ As mentioned, there are registry and remote modules. Both types of modules are below. By this point throughout the Terraform chapters, you may or may not have a VPC in the AWS account you are deploying to. The first exercise creates an ec2 instance and VPC in AWS. Alternatively you may want to add an EC2 instance to an existing VPC, that is also covered below. 
+
+
 ## Exercise
 
-TODO
+### **Registry Modules** 
+
+There are a multitude of registry modules available. For this exercise the modules used are managed by hashicorp. 
+
+?> Whenever you add a new module to a configuration, Terraform must install the module before it can be used. Both the `terraform get` and `terraform init` commands will install and update modules.
+
+
+
+1. Create a `variables.tf` file and add the following variables. If you already have a `variables.tf` file, add them there. Note that there are several variables that should be updated as they should be specific to your stack, variables such as `vpc_name` should be updated. Update the `default` parameter 
+
+```hcl
+variable "vpc_name" {
+  description = "Name of VPC"
+  type        = string
+  default     = "<your-vpc-name>"
+}
+
+variable "vpc_cidr" {
+  description = "CIDR block for VPC"
+  type        = string
+  default     = "10.0.0.0/16"
+}
+
+variable "vpc_azs" {
+  description = "Availability zones for VPC"
+  type        = list(string)
+  default     = ["us-east-2a", "us-east-2b", "us-east-2c"]
+}
+
+variable "vpc_private_subnets" {
+  description = "Private subnets for VPC"
+  type        = list(string)
+  default     = ["10.0.1.0/24", "10.0.2.0/24"]
+}
+
+variable "vpc_public_subnets" {
+  description = "Public subnets for VPC"
+  type        = list(string)
+  default     = ["10.0.101.0/24", "10.0.102.0/24"]
+}
+
+variable "vpc_enable_nat_gateway" {
+  description = "Enable NAT gateway for VPC"
+  type        = bool
+  default     = true
+}
+
+variable "vpc_tags" {
+  description = "Tags to apply to resources created by VPC module"
+  type        = map(string)
+  default = {
+    Terraform   = "true"
+    Environment = "sandbox"
+  }
+}
+```
+
+2. Review the configuration below first. The `  source  = "terraform-aws-modules/vpc/aws"
```

Comment:
Potentially worth adding a link to documentation for these modules and a note than when referencing public modules it's important to review the documentation to know how the modules can be configured and what they create.

Reply:
Great callout. Added 👍